docs/rules/object-key-newlines.md

```diff
@@ -0,0 +1,68 @@
+# Enforce placing object keys on separate lines (object-key-newlines)
+
+While formatting preferences are very personal, a number of style guides require that
```

Comment:
Sorry, that was my bad. I asked him to remove code because it was so similar to the correct example.

My reasoning has been code in intro helps to illustrate JavaScript syntax or idiom that is unfamiliar to many readers, but if common like object literal notation, and few options, prioritize getting some example code “above the fold”

Agree that “why” or “what problem does it solve” is a top priority in the introduction between main h1 heading and h2  Rule Details. Because the issue does not provide, some research needed. As a starting point, it is my impression that this style (with the optional exception) is common in React.